File: day6.py
from collections import Counter

with open("input6.txt") as f:
    nums = [int(n) for n in f.read().strip().split(",")]


track = Counter(nums)

for day in range(256):
    new = track[0]
    for i in range(8):
        track[i] = track[i + 1]
    track[8] = new
    track[6] += new


print(sum(track.values()))

File: day6.jl
f = open("input6.txt")
inp = read(f, String)
close(f)

inp = [parse(Int32, i) for i in split(inp,',')]
track = Dict((i, 0) for i in 0:8)

for i in inp
    track[i] += 1
end

for i in 1:256
    track[(i+6)%9] += track[(i-1)%9]
    
end

println(sum(values(track)))